app/controllers/workflow/pages_controller.rb

```diff
@@ -26,13 +26,55 @@ def request_approval
         args = { f_uid: @cur_user._id, t_uid: workflow_approver[:user_id],
                  site: @cur_site, page: @item,
                  url: params[:url], comment: params[:workflow_comment] }
-        Workflow::Mailer.request_mail(args).deliver_now
+        Workflow::Mailer.request_mail(args).deliver_now if mail_enabled?(args[:t_uid])
       end
 
       @item.set_workflow_approver_state_to_request
       @item.update
     end
 
+    def mail_enabled?(target_id)
+      target_user = SS::User.find(target_id) rescue false
+      target_user_email = target_user.email if target_user.present?
+      @cur_user.email.present? && target_user_email.present?
+    end
+
+    def email_blank_ids
+      email_blank_id = []
+      email_blank_id.push(@cur_user._id) if @cur_user.email.blank?
+      if @item.workflow_state == "request"
+        current_level = @item.workflow_current_level
+        current_workflow_approvers = @item.workflow_approvers_at(current_level)
+        current_workflow_approvers.each do |workflow_approver|
+          if @cur_user._id !=workflow_approver[:user_id]
+            approver_user = SS::User.where(id: workflow_approver[:user_id]).first
+            if approver_user
+              approver_user_email = approver_user.email
+              email_blank_id.push(approver_user._id) if approver_user_email.blank?
```

Comment:
__[rubocop]__
* Avoid more than 3 levels of block nesting. [:ref](https://github.com/bbatsov/ruby-style-guide#three-is-the-number-thou-shalt-count)

<!-- Comment by [SideCI](https://sideci.com) -->